NorthstarDLL/masterserver.cpp

```diff
@@ -221,24 +294,11 @@ void MasterServerManager::RequestServerList()
 
 				for (auto& serverObj : serverArray)
 				{
-					if (!serverObj.IsObject())
-					{
-						spdlog::error("Failed reading masterserver response: member of server array is not an object");
-						goto REQUEST_END_CLEANUP;
-					}
-
-					// todo: verify json props are fine before adding to m_remoteServers
-					if (!serverObj.HasMember("id") || !serverObj["id"].IsString() || !serverObj.HasMember("name") ||
-						!serverObj["name"].IsString() || !serverObj.HasMember("description") || !serverObj["description"].IsString() ||
-						!serverObj.HasMember("map") || !serverObj["map"].IsString() || !serverObj.HasMember("playlist") ||
-						!serverObj["playlist"].IsString() || !serverObj.HasMember("playerCount") || !serverObj["playerCount"].IsNumber() ||
-						!serverObj.HasMember("maxPlayers") || !serverObj["maxPlayers"].IsNumber() || !serverObj.HasMember("hasPassword") ||
-						!serverObj["hasPassword"].IsBool() || !serverObj.HasMember("modInfo") || !serverObj["modInfo"].HasMember("Mods") ||
-						!serverObj["modInfo"]["Mods"].IsArray())
+					if (!serverObj.Accept(validator))
 					{
 						spdlog::error("Failed reading masterserver response: malformed server object");
```

Comment:
Is it possible to have a more descriptive error here?

Reply:
What sort of thing would you want to be mentioned?

Reply:
Ugh, I suppose what I was hoping for doesn't really exist: https://github.com/Tencent/rapidjson/issues/551.